src/instance/read.jl

```diff
@@ -98,6 +98,10 @@ function _from_json(json; repair = true)
         json["Parameters"]["Power balance penalty (\$/MW)"],
         default = [1000.0 for t in 1:T],
     )
+    shortfall_penalty = timeseries(
+        json["Parameters"]["Reserve shortfall penalty (\$/MW)"],
+        default = [0.0 for t in 1:T],
```

Comment:
For consistency with other penalties, `1000.0` might be a better default. I'm also not sure about using `0.0` to indicate "constraint must always be satisfied". Note that we don't have this convention for other penalties.

Reply:
With ea35c3f (and docs corrected in 2cdf887), default is set to -1, indicating that constraint must always be satisfied.

Reply:
- My suggestion was to make this constraint soft by default, with a penalty of `1000.0`, to keep it consistent with the other penalties we already have in the input file. I want to avoid some penalties being infinity by default and others being a fixed number.
- In the JSON format, I think `"Infinity"` would be more clear than `-1`, but we can keep `-1` for now.

Reply:
> * My suggestion was to make this constraint soft by default, with a penalty of `1000.0`, to keep it consistent with the other penalties we already have in the input file. I want to avoid some penalties being infinity by default and others being a fixed number.

It is up to you. I personally would avoid adding extra variables to a formulation when they are unnecessary. This is a bit harder for a preprocessor to figure out, and I am not familiar enough with the scales of the input data to know whether a penalty of `1000.0` would guarantee that the constraint is enforced.

> * In the JSON format, I think `"Infinity"` would be more clear than `-1`, but we can keep `-1` for now.

That makes sense.

Reply:
Let's keep `-1` as default for now. I will check if it's common for the ISOs to use soft reserve constraints.